src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/api/ReportRunController.kt

```diff
@@ -0,0 +1,82 @@
+package org.vaccineimpact.orderlyweb.controllers.api
+
+import com.google.gson.Gson
+import org.vaccineimpact.orderlyweb.ActionContext
+import org.vaccineimpact.orderlyweb.OrderlyServer
+import org.vaccineimpact.orderlyweb.OrderlyServerAPI
+import org.vaccineimpact.orderlyweb.controllers.Controller
+import org.vaccineimpact.orderlyweb.db.AppConfig
+import org.vaccineimpact.orderlyweb.db.Config
+import org.vaccineimpact.orderlyweb.db.repositories.OrderlyWebReportRunRepository
+import org.vaccineimpact.orderlyweb.db.repositories.ReportRunRepository
+import java.time.Instant
+
+data class ReportRun(val name: String, val key: String, val path: String)
+
+class ReportRunController(
+    context: ActionContext,
+    private val reportRunRepository: ReportRunRepository,
+    private val orderlyServerAPI: OrderlyServerAPI,
+    config: Config
+) : Controller(context, config)
+{
+    constructor(context: ActionContext) :
+            this(
+                context,
+                OrderlyWebReportRunRepository(),
+                OrderlyServer(AppConfig()),
+                AppConfig()
+            )
+
+    fun run(): String
+    {
+        val name = context.params(":name")
+
+        val instances: Map<String, String> = context.postData("instances")
+        val params: Map<String, String> = context.postData("params")
+        val gitBranch: String = context.postData("gitBranch")
+        val gitCommit: String = context.postData("gitCommit")
+
+        val response =
+            orderlyServerAPI.post(
+                "/v1/reports/$name/run/",
+                Gson().toJson(params),
+                mapOf(
+                    "ref" to gitCommit,
+                    // TODO remove this in favour of passing instances itself to orderly.server - see VIMC-4561
+                    "instance" to when (instances.isEmpty())
+                    {
+                        true -> ""
+                        false -> instances.values.iterator().next()
+                    }
+                )
+            )
+        val reportRun = response.data(ReportRun::class.java)
```

Comment:
I guess you only want to attempt adding to the db if response has a success status

Reply:
I've added the throw-on-error parameter

Reply:
Oh, interesting. @Nicolas-Dolan is doing something similar in his branch and is just passing through the first response if it's not successful. We should be consistent here really - i suppose the passthrough gives back more information about the error, and it's what we're doing for most single calls. But we must be throwing that OrderlyServerError elsewhere (when constructing the run report page, and in the bundle controller). 

I wonder if we should use OrderlyServerError more universally, but add some more information to the exception so that we log it. What do you think?

Reply:
Yes, to using `OrderlyServerError` more broadly. I think the alternative should be the exception - it's too easy to forget to handle an error. Are you ok with me opening a maintenance ticket where I try to explain the issue and perhaps you can add any thoughts? I'm just aware that this PR is fairly big and blocking some other tickets.

Reply:
Good idea

Reply:
https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-2183